src/main/java/com/exactpro/th2/mstore/ErrorCollector.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2023 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.exactpro.th2.mstore;
+
+import com.exactpro.th2.common.event.Event;
+import com.exactpro.th2.common.event.Event.Status;
+import com.exactpro.th2.common.event.IBodyData;
+import com.exactpro.th2.common.grpc.EventBatch;
+import com.exactpro.th2.common.grpc.EventID;
+import com.exactpro.th2.common.schema.message.MessageRouter;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonFormat;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.datatype.jsr310.ser.InstantSerializer;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import static java.util.Objects.requireNonNull;
+
+@SuppressWarnings("unused")
+public class ErrorCollector implements AutoCloseable {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ErrorCollector.class);
+    private static final ThreadFactory THREAD_FACTORY = new ThreadFactoryBuilder().setNameFormat("error-collecor-%d").build();
+    private final ScheduledFuture<?> drainFuture;
+    private final MessageRouter<EventBatch> eventRouter;
+    private final EventID rootEvent;
+    private final Lock lock = new ReentrantLock();
+    private Map<String, ErrorMetadata> errors = new HashMap<>();
```

Comment:
Why don't make it final an simply call `clear` method when we want to clear the collected errors?

Reply:
I copy this map to create event outside the lock. New map is created in this case. I use this approach to avoid redundant 
clean operation. Please take look at the clear method of this class

Reply:
Okay, thank you for the explanation